ecosystem/typescript/sdk/src/aptos_types/type_tag.ts

```diff
@@ -386,6 +410,15 @@ export class TypeTagParser {
         bail("Invalid type tag.");
       }
 
+      // an Object `0x1::object::Object<T>` doesn't hold a real type, it points to an address
+      // therefore, we parse it as an address and dont need to care/parse the `T` type
+      if (module === "object" && name === "Object") {
+        // to support a nested type tag, i.e 0x1::some_module::SomeResource<0x1::object::Object<T>>, we want
+        // to remove the `<T>` part from the tokens list so we dont parse it and can keep parse the type tag.
+        this.tokens.splice(0, 3);
```

Comment:
just wanna make sure that we don't one day encounter a materialized Object type like `0x4::object::Object<0x1::aptos_coin::AptosCoin>`? It is always going to be `0x4::object::Object<T>`?

Reply:
yes. verified it before with @gerben-stavenga